### PR TITLE
Move http module to top-level module

### PR DESCRIPTION
**Pull request by jeffbcross**
90% chance this is going to fail on Travis

### CODE REVIEW

**Comment by jeffbcross** [comment]
Latest force-push will probably fix the two previously-failing Travis jobs.

**Comment by jeffbcross** [comment]
I still have a few todos to cleanup and test docs, as well as check that our code.angularjs.org release step is creating a proper bundle.

File: docs/typescript-definition-package/templates/type-definition.template.html [hunk 1]
```diff
@@ -67,6 +67,6 @@
 
 {% endfor %}
 
-declare module "angular2/angular2" {
+declare module "{$ doc.module $}" {
```
Comment (mhevery) [inline docs/typescript-definition-package/templates/type-definition.template.html]: I don't think this is that simple.

File: gulpfile.js [hunk 2]
```diff
@@ -615,7 +615,7 @@ gulp.task('test.unit.dart', function (done) {
         return;
       }
 
-      watch('modules/angular2/**', { ignoreInitial: true }, [
+      watch(['modules/angular2/**', 'modules/angular2_http/**'], { ignoreInitial: true }, [
```
Comment (mhevery) [inline gulpfile.js]: agree with this strategy for naming.

File: modules/angular2_http/angular2_http.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
+/**
```
Comment (mhevery) [inline modules/angular2_http/angular2_http.ts]: rename to `angular2_http/http`

**Comment by jeffbcross** [comment]
Per in-person review with @mhevery, I'm going to explore simplifying the package name to 'http', imported as angular2/http, and published to npm under the angular2 namespace to be installed as '@angular2/http' (the "@" prefix is unavoidable for npm namespaces). 

For this reason, I believe for the typescript definition to correctly transpile import statements, the module would need to be declared as "angular2/http". Not ideal. I'm doing research to see what the options are.

**Comment by jeffbcross** [comment]
Also, since pub apparently doesn't support namespaces, I think we'd have to keep the angular2_http name for that.

**Comment by jeffbcross** [comment]
@mhevery I think the reason I named the main export file angular2_http is because `pub publish` gives a warning if there is not a library defined with a filename that matches the package name. https://www.dartlang.org/tools/pub/package-layout.html

**Comment by mhevery** [comment]
There is no rule which says that Dart and NPM need to fallow the same
naming strategy. We can diverge or ignore the warning.

**Comment by jeffbcross** [comment]
I'm writing up an issue to discuss module publishing and naming strategy, so we can discuss and decide during our team meeting.

**Comment by mhevery** [comment]
Sgtm

**Comment by jeffbcross** [comment]
@mhevery I opened an issue to discuss module publishing semantics in #3458.

Implications for this PR:
 * I will not publish http as a separate dart package, will just include it in angular2. 
 * Http will have its own bundle, http.js. 
 * Http will be published to npm as part of angular2 (`require('angular2/http')`), as well as its own module called "ngHttp" (`require('ngHttp')`), and under the angular namespace as "@angular/http" (`npm install @angular/http`, `require('@angular/http')`).

So the only changes I'd make based on these changes is name changes and changes to our release script.

**Comment by mhevery** [comment]
SGTM.

**Comment by jeffbcross** [comment]
Submitted this to a presubmit branch because the only failure is a flake that's already occurring on master.

**Pull request closed without merging**